### PR TITLE
Pre-tokenize, DDP and eval metrics

### PR DESCRIPTION
**Pull request by devWhyqueue**
This PR introduces several significant updates to the fine-tuning process with the Civil Comments dataset:

## Key Changes
1. **Pre-tokenized Dataset**:  
   - Introduced a new CLI command, `tokenize`, to pre-tokenize datasets for efficient processing during fine-tuning.

2. **PyTorch Distributed Data Parallel (DDP)**:  
   - Implemented DDP with 8 GPUs to speed up fine-tuning.

3. **Evaluation Metrics Update**:  
   - Adapted evaluation metrics to align with those used in the baseline for better comparability.

## Fine-tuning Details
- **Dataset**: Civil Comments  
- **Batch Size**: 16  
- **Epochs**: 1 
- **Runtime**: Approximately 4 hours  

### Evaluation Results
- **Loss**: 0.2256  
- **Classification Report**:
  ```

              precision    recall  f1-score   support

           0       0.99      0.98      0.98     91532
           1       0.69      0.76      0.73      5788

    accuracy                           0.97     97320
   macro avg       0.84      0.87      0.85     97320
  weighted avg       0.97      0.97      0.97     97320

  ```
- **ROC-AUC Score**: 0.9808  

### Weights

You can find the fine-tuned weights under `/home/space/datasets/civil_comments/weights` on the cluster.

### CODE REVIEW

**Comment by devWhyqueue** [comment]
Issue resolved. Evaluation has to be distributed, too. Started a new fine-tuning run and waiting for results...

**Review by Imenee98 — APPROVED** [review]
